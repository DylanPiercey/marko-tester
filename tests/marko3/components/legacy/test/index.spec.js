const { render, fixtures } = require('../../../../..')('../../../../marko4/components/legacy/component.js');

beforeEach(() => {
  window.alert = jest.fn();
});

describe('When legacy component is rendered with data', () => {
  let component;

  beforeEach(() => {
    component = render(fixtures.index);
  });

  afterEach(() => {
    component.destroy();
  });

  it('should init legacy component', () => {
    expect(window.alert).toBeCalledWith(fixtures.index.visible);
  });

  describe('When user clicks the button', () => {
    beforeEach((done) => {
      component.getEl('button').click();
      component.once('update', done);
    });

    it('should set change the state and re-render the template', () => {
      expect(component.state.visible).toBe(false);
    });
  });
});
