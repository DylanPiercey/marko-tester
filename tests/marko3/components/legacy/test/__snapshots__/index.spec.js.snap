// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component with index fixture 1`] = `
Array [
  <div
    class="legacy"
    id="wc0"
  >
    <section>
      Section
    </section>
    <button
      data-w-onclick="changeInput|wc0"
      id="wc0-button"
      type="button"
    >
      Hello
    </button>
    <inner-legacy />
  </div>,
]
`;
